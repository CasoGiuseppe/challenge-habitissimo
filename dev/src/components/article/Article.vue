<template>
  <article
    :class="[
      'article',
      arrow ? `article--hasArrow-${arrow.direction}`:null
    ]">
      <h3
        v-if="category"
        class="article__category">
        <transition
          mode="out-in"
          name="change-move">
            <span
              :key="category">
              {{category}}
            </span>
        </transition>
      </h3>
      <h2
        v-if="$slots.title"
        class="article__title"
        v-html="$slots.title[0].text">
      </h2>
      <p
        v-if="$slots.description"
        class="article__description"
        v-html="$slots.description[0].text">
      </p>
      <div
        v-if="$slots.extra"
        class="article__extra"
        v-html="$slots.extra[0].text">
      </div>
      <div
        v-if="$slots.action"
        class="article__action">
        <slot name="action" />
      </div>
  </article>
</template>

<script>
export default {
  name: 'Article',

  props: {
    category: {
      type: String,
    },

    arrow: {
      type: Object,
    },
  },
};
</script>

<style lang='scss' scoped src= './Article.scss'></style>
