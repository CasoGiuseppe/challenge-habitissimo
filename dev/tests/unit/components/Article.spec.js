import Article from '@/components/article/Article'
import { mount } from '@vue/test-utils'

describe('Article.vue', () => {
  let component

  beforeEach(() => {
    component = mount(Article, {
      propsData:{
        category: 'cat',
        arrow: {
          'direction': 'dir'
        }
      },
      slots: {
        action: '<span class="slot-action">title</span>'
      }
    })
  })

  it('has correct action slot', () => {
    const main = component.find('.article')
    expect(main.findAll(".slot-action").length).toBe(1);
  })

  it('has correct props values', () => {
    expect(component.props().category).toBe('cat')
    expect(component.props().arrow).toHaveProperty('direction', 'dir')
  })

  it('has the expected html structure', () => {
    expect(component.html()).toMatchSnapshot()
  })
})