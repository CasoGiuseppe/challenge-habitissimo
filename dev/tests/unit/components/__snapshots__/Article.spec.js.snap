// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article.vue has the expected html structure 1`] = `
<article class="article article--hasArrow-dir">
  <h3 class="article__category">
    <transition-stub name="change-move" mode="out-in"><span>
            cat
          </span></transition-stub>
  </h3>
  <!---->
  <!---->
  <!---->
  <div class="article__action"><span class="slot-action">title</span></div>
</article>
`;
